tests/components/bottom_bar.test.ts

```diff
@@ -466,4 +468,80 @@ describe("BottomBar component", () => {
     expect(fixture.querySelector(".o-selection-statistic")?.textContent).toBe("Count Numbers: 1");
     app.destroy();
   });
+
+  describe("drag&drop sheet", () => {
+    let sheetIds;
+    let model;
+    let app;
+
+    beforeAll(async () => {
+      sheetIds = ["Sheet1", "Sheet2", "Sheet3", "Sheet4"];
+      const originalGetBoundingClientRect = HTMLDivElement.prototype.getBoundingClientRect;
+      jest
+        .spyOn(HTMLDivElement.prototype, "getBoundingClientRect")
+        // @ts-ignore the mock should return a complete DOMRect, not only { top, left }
+        .mockImplementation(function (this: HTMLDivElement) {
+          for (let sheetId of sheetIds) {
+            if (this.dataset.id == sheetId) {
+              return { top: 100, left: (sheetIds.indexOf(sheetId) + 1) * 100 };
+            }
+          }
+          return originalGetBoundingClientRect.call(this);
+        });
+    });
+
+    beforeEach(async () => {
+      model = new Model({
+        sheets: sheetIds.map((sheetId) => {
+          name: sheetId;
+        }),
+      });
+      ({ app } = await mountSpreadsheet(fixture, { model }));
+    });
+    afterEach(async () => {
+      app.destroy();
+    });
+    test("Can drag&drop a sheet forward", async () => {
+      dragSheet(sheetIds[0], { x: 1000 });
+      expect(model.getters.getVisibleSheetIds()).toEqual(["Sheet2", "Sheet3", "Sheet4", "Sheet1"]);
+    });
+    test("Can drag&drop a sheet backward", async () => {
+      dragSheet(sheetIds[2], { x: 10 });
+      expect(model.getters.getVisibleSheetIds()).toEqual(["Sheet3", "Sheet1", "Sheet2", "Sheet4"]);
+    });
+    test("Can drag&drop a sheet on itself", async () => {
```

Comment:
Add maybe a test where the mouseMove/Up are on some random element, and not on the sheet element itself.